test/ecto_state_machine_test.exs

```diff
@@ -95,6 +95,29 @@ defmodule EctoStateMachineTest do
     end
   end
 
+  describe "#validate_rules_change" do
+    test "valid state change", context do
+      changeset = context[:confirmed_user]
+                  |> Ecto.Changeset.change(%{rules: "admin"})
+                  |> User.validate_rules_change
+      assert changeset.valid? == true
+    end
+
+    test "invalid state change", context do
```

Comment:
Please test all of the possible transitions. It's a little bit paranoic, but it's the testing convention right now